read.go

```diff
@@ -226,24 +230,22 @@ func readNativeFrames(d dicomio.Reader, parsedData *Dataset, fc chan<- *frame.Fr
 		buf := make([]int, int(pixelsPerFrame)*samplesPerPixel)
 		for pixel := 0; pixel < int(pixelsPerFrame); pixel++ {
 			for value := 0; value < samplesPerPixel; value++ {
-				if bitsAllocated == 8 {
-					val, err := d.ReadUInt8()
-					if err != nil {
-						return nil, bytesRead, err
+
+				n, err := d.Read(pixelBuf)
+				if err != nil || n < bytesAllocated {
+					if err == nil {
+						err = ErrorIncompleteRead
 					}
-					buf[(pixel*samplesPerPixel)+value] = int(val)
+					return nil, bytesRead,
+						fmt.Errorf("could not read uint%d from input: %w", bitsAllocated, err)
+				}
```

Comment:
Do you think this would read cleaner as two if statements instead of a nested if?

```go
if n < bytesAllocated {
  return nil, bytesRead, ErrorIncompleteRead
}

if err != nil {
    return nil, bytesRead, fmt.Errorf(...)
}
```

Reply:
I think we have to get rid of first condition since `err` is either `EOF` or `ErrUnexpectedEOF`

Reply:
Yeah, let's maybe just wrap and pass the error up in this case.